### PR TITLE
net: dhcpv4: Reimplement RENEW/REBIND logic according to RFC2131

### PR DESCRIPTION
**Pull request by rlubos**
DHCP Request retransmission in RENEW and REBIND states was not compliant with RFC2131.

The retransmission interval should not be calculated as in REQUESTING state in such case, but rather calculated based on the remaining T2 or lease time (depending on current state).

RFC doesn't also mention any retransmission count limit for those states. The client should retransmit the Request until T2 or lease expiry time is reached.

### CODE REVIEW

File: subsys/net/lib/dhcpv4/dhcpv4.c [hunk 1]
```diff
@@ -397,6 +397,64 @@ static void dhcpv4_set_timeout(struct net_if_dhcpv4 *dhcpv4,
 	k_work_reschedule(&timeout_work, K_NO_WAIT);
 }
 
+/* Set a new timeout w/o updating base time. Used for RENEWING and REBINDING to
+ * keep track of T1/T2/lease timeouts.
+ */
+static void dhcpv4_set_timeout_inc(struct net_if_dhcpv4 *dhcpv4,
+				   int64_t now, uint32_t timeout)
+{
+	int64_t timeout_ms;
```
Comment (jukkar) [inline subsys/net/lib/dhcpv4/dhcpv4.c:406]: I just wonder, would it suffice to use uint32 for the current time?

File: subsys/net/lib/dhcpv4/dhcpv4.c [hunk 2]
```diff
@@ -660,36 +718,60 @@ static void dhcpv4_enter_requesting(struct net_if *iface, struct dhcp_msg *msg)
...
+		dhcpv4->renewal_time = dhcpv4->lease_time / 2U;
 	}
 
-	rebinding_time = iface->config.dhcpv4.rebinding_time;
-	if (!rebinding_time) {
+	if (dhcpv4->rebinding_time == 0) {
 		/* The default rebinding time rfc2131 4.4.5 */
-		rebinding_time = iface->config.dhcpv4.lease_time * 875U / 1000;
-		iface->config.dhcpv4.rebinding_time = rebinding_time;
+		dhcpv4->rebinding_time = dhcpv4->lease_time * 875U / 1000;
```
Comment (jukkar) [inline subsys/net/lib/dhcpv4/dhcpv4.c:731]: Makes probably no difference but we should ise 1000U here, also noticed similar things in other functions

**Review by jukkar — DISMISSED** [review]
Looks good, minor nits.

File: subsys/net/lib/dhcpv4/dhcpv4.c (hunk 2 above)
Comment (rlubos) [reply subsys/net/lib/dhcpv4/dhcpv4.c:731]: I'll look over the diff and fix where applicable.

Comment (rlubos) [reply subsys/net/lib/dhcpv4/dhcpv4.c]: Should be fixed now.

**Review by jukkar — APPROVED** [review]

**Review by maass-hamburg — APPROVED** [review]

**Pull request merged**